flatpak/com.github.Alcaro.Flips.json

```diff
@@ -0,0 +1,37 @@
+{
+    "app-id" : "com.github.Alcaro.Flips",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.34",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flips",
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc",
+        "--socket=x11",
+        /* Wayland access */
+        "--socket=wayland",
+        /* OpenGL + DRI access */
+        "--device=dri",
+        /* Needed to find the ROMs and patches */
+        "--filesystem=host:ro"
+    ],
+    "modules" : [
+        {
+            "name" : "flips",
+            "no-autogen" : true,
+            "builddir" : true,
+            "make-args": [
+                "prefix=/app"
+            ],
```

Comment:
OK, ping me once it's possible and I'll rebase. :)

FYI, the SDK sets some flags by default: https://github.com/flatpak/freedesktop-sdk-images/blob/1.6/org.freedesktop.Sdk.json.in#L110.

Reply:
Those flags look like a good start, but make.sh is a lot more aggressive than that, including profile-guided optimization.

The fix is up. Feel free to check which other flags make.sh accepts, or suggest more; I usually optimize for binary size, but judging by the -g, you may want --strip=no.

Reply:
MR adding the `application/x-ips-patch` MIME type: https://gitlab.freedesktop.org/xdg/shared-mime-info/merge_requests/53.

Reply:
Seconding the citation needed on International Patching System.

I could swear [this guy](https://www.smwcentral.net/?p=postsbyuser&uid=9435) said Internal Patching System at some point (he's been around the ROM hacking scene since approximately before it even existed, he knows what he's doing), but I can't find the post.

I think BPS is Binary Patching System, but I'm having [a bit of trouble](https://web.archive.org/web/20000101010101/http://byuu.org) finding a citation for that too.

Reply:
I assumed it was Byuu Patching System (really I have no idea).

Reply:
Fair chance he's trying to retcon it into beat patching system (unknown capitalization). I'll ask next time I find him.

Either way, [I found a citation for IPS](https://smwc.me/1403322), though I'm not sure if it meets Wikipedia or FDo standards. I can swear I saw an in-depth discussion about the origins of IPS somewhere, including which tool invented it (I believe it was a now-dead DOS-based emulator), but I can't find it.